src/syscheckd/db/src/file.cpp

```diff
@@ -1,537 +1,254 @@
 /**
- * @file fim_db_files.c
+ * @file file.cpp
  * @brief Definition of FIM database for files library.
- * @date 2020-09-9
+ * @date 2021-09-9
  *
  * @copyright Copyright (C) 2015-2021 Wazuh, Inc.
  */
+#include "json.hpp"
+#include "db.hpp"
+#include "fimDBHelper.hpp"
+#include "dbFileItem.hpp"
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#include "db.hpp"
-
-#ifdef WAZUH_UNIT_TESTING
-/* Remove static qualifier when unit testing */
-#define static
-
-/* Replace assert with mock_assert */
-extern void mock_assert(const int result, const char* const expression, const char* const file, const int line);
-#undef assert
-#define assert(expression) mock_assert((int)(expression), #expression, __FILE__, __LINE__);
-#endif
-
-extern const char* SQL_STMT[];
-
-// Convenience macros
-#define fim_db_bind_set_scanned(fim_sql, path) fim_db_bind_path(fim_sql, FIMDB_STMT_SET_SCANNED, path)
-
-#define fim_db_bind_get_path_from_pattern(fim_sql, path) \
-    fim_db_bind_path(fim_sql, FIMDB_STMT_GET_PATH_FROM_PATTERN, path)
-
-// bindings
-
+const auto fileColumnList = R"({"column_list":"[path, mode, last_event, scanned, options, checksum, dev, inode, size,
+                                                perm, attributes, uid, gid, user_name, group_name, hash_md5, hash_sha1,
+                                                hash_sha256, mtime]"})"_json;
 
 /**
- * @brief Binds a range of paths.
+ * @brief Get path list using the sqlite LIKE operator using @pattern. (stored in @file).
+ * @param pattern Pattern that will be used for the LIKE operation.
  *
- * @param fim_sql FIM database structure.
- * @param file_path File name of the file to insert.
- * @param entry FIM entry data structure.
+ * @return a vector with every paths on success, a empty vector otherwise.
  */
-static void fim_db_bind_replace_entry(fdb_t* fim_sql, const char* file_path, const fim_file_data* entry);
-
+std::vector<std::string> fim_db_get_path_from_pattern(const char *pattern);
```

Comment:
Move this out of the __cplusplus, save or avoid it. These are C++ definitions in the implementation of a C interface

Reply:
use camelCase